Modules/Core/Common/include/itkCellInterface.h

```diff
@@ -54,27 +54,27 @@
-  using typename Superclass::CellRawPointer;                                                                           \
-  using typename Superclass::CellConstRawPointer;                                                                      \
-  using typename Superclass::CellTraits;                                                                               \
-  using typename Superclass::CoordRepType;                                                                             \
-  using typename Superclass::InterpolationWeightType;                                                                  \
-  using typename Superclass::PointIdentifier;                                                                          \
-  using typename Superclass::PointIdIterator;                                                                          \
-  using typename Superclass::PointIdConstIterator;                                                                     \
-  using typename Superclass::CellIdentifier;                                                                           \
-  using typename Superclass::CellFeatureIdentifier;                                                                    \
+  using PixelType = typename Superclass::PixelType;                                                                    \
+  using CellType = typename Superclass::CellType;                                                                      \
+  using CellAutoPointer = typename Superclass::CellAutoPointer;                                                        \
+  using CellConstAutoPointer = typename Superclass::CellConstAutoPointer;                                              \
+  using CellRawPointer = typename Superclass::CellRawPointer;                                                          \
+  using CellConstRawPointer = typename Superclass::CellConstRawPointer;                                                \
+  using CellTraits = typename Superclass::CellTraits;                                                                  \
+  using CoordRepType = typename Superclass::CoordRepType;                                                              \
+  using InterpolationWeightType = typename Superclass::InterpolationWeightType;                                        \
+  using PointIdentifier = typename Superclass::PointIdentifier;                                                        \
+  using PointIdIterator = typename Superclass::PointIdIterator;                                                        \
+  using PointIdConstIterator = typename Superclass::PointIdConstIterator;                                              \
+  using CellIdentifier = typename Superclass::CellIdentifier;                                                          \
+  using CellFeatureIdentifier = typename Superclass::CellFeatureIdentifier;                                            \
```

Comment:
Are you sure all of these changes are necessary, for VS2017? I had the impression that the VS2017 error "C2886: symbol cannot be used in a member using-declaration" only occurred when the "using type" itself a base class of this class. (Compiler bug https://developercommunity.visualstudio.com/t/486683)

Reply:
You are more than welcome to take over this PR, and explore the minimum amount of changes which get it compile on VS2017.

Reply:
I'm having a look now, and it appears that there's another category of erroneous (!) VS2017 compile errors that we're encountering now after pull request #2567, saying:

> error C2244: unable to match function definition to an existing declaration

This VS2017 compile error may be reproduced by:

```
template <typename T>
class Base
{
public:
  using NestedType = T;
};

template <typename T>
class Derived : public Base<T>
{
  using Superclass = Base<T>;
  using typename Superclass::NestedType;
  NestedType MemberFunction();
};

template <typename T>
typename Derived<T>::NestedType
Derived<T>::MemberFunction()
{  // <==  error C2244: unable to match function definition to an existing declaration

  return NestedType{};
}
```

And this particular error may be solved by a _trailing return type_ instead:

```
template <typename T>
auto
Derived<T>::MemberFunction() -> NestedType
{  // OK!

  return NestedType{};
}
```

And in this particular case (having a nested dependent type as return type), I think it would actually be quite nice to use a trailing return type.

To be continued...